### PR TITLE
DO NOT MERGE - Stop sending empty files to EBI

### LINKED ISSUE #1070: Modify EBI code so that it does not upload empty files
These cause error messages to be thrown from EBI's end

### PR DESCRIPTION
**Pull request by adamrp**
Still need to test this out, but I think this will do the trick

Fix #1070

### CODE REVIEW

File: qiita_ware/ebi.py [hunk 1]
```diff
@@ -20,6 +21,11 @@
 from qiita_db.util import convert_to_id
 
 
+class EmptyFileWarning(Warning):
```
Comment (josenavas) [inline qiita_ware/ebi.py]: I recommend to add this warning in the exceptions class and add a filter so the Warning is always raised. There is an example in qiita_db.exceptions.

**Comment by josenavas** [comment]
Is this really the best way of checking for empty files? I think it will be better if when we are generating the per sample fastq we detect that a sample do not have sequences and we don't write that file out. Does that makes sense?

File: qiita_ware/ebi.py [hunk 2]
```diff
@@ -404,6 +410,17 @@ def add_sample_prep(self, sample_name, platform, file_type, file_path,
                              "%s (case insensitive)" % (platform,
                                                         ', '.join(platforms)))
 
+        with open(file_path) as fp:
```
Comment (wasade) [inline qiita_ware/ebi.py]: what about:

```python
import os
if not os.path.exists(file_path) or os.stat(file_path).st_size == 0:
    # warn, return
```

**Comment by wasade** [comment]
@josenavas, it may depend on how that code is written as to if it is easier

**Comment by adamrp** [comment]
@josenavas yes, I think that would be better, but I thought the per-sample fastq files were being generated by QIIME.

@wasade I think using `os.stat(file_path).st_size` would be fine.

File: qiita_ware/ebi.py (hunk 1 above)
Comment (adamrp) [inline qiita_ware/ebi.py]: I'll move the warning to `qiita_ware.exceptions` and import it here. What is the purpose of the filter? i.e., in what circumstances would the warning not be raised if the `warnings.simplefilter("always")` were not there?

**Comment by josenavas** [comment]
@adamrp if you don't add the filter, in the live system the warning will be raised only once, (the first time), since this is the default behavior for warnings in python.

**Comment by adamrp** [comment]
Okay, thanks! I added the filter.

**Comment by adamrp** [comment]
Actually, checking st_size == 0 is not sufficient, since a gzipped empty file is not zero bytes:

```
(python276)adam:test@ebi_empty_files-issue-1070$ touch hello
(python276)adam:test@ebi_empty_files-issue-1070$ du -h hello
  0B	hello
(python276)adam:test@ebi_empty_files-issue-1070$ gzip hello
(python276)adam:test@ebi_empty_files-issue-1070$ du -h hello.gz
4.0K	hello.gz
```

**Comment by adamrp** [comment]
And not all empty files have the same md5 (the name of the file is encoded):

```
(python276)adam:test@ebi_empty_files-issue-1070$ touch test test2
(python276)adam:test@ebi_empty_files-issue-1070$ gzip test test2
(python276)adam:test@ebi_empty_files-issue-1070$ md5 test.gz test2.gz
MD5 (test.gz) = 6976ece38ea13638c2c4724da1fdd640
MD5 (test2.gz) = a84c25cf07c3c8fbd22e6c6d69c3f615
(python276)adam:test@ebi_empty_files-issue-1070$ xxd test.gz
0000000: 1f8b 0808 684a 4d55 0003 7465 7374 0003  ....hJMU..test..
0000010: 0000 0000 0000 0000 00                   .........
(python276)adam:test@ebi_empty_files-issue-1070$ xxd test2.gz
0000000: 1f8b 0808 684a 4d55 0003 7465 7374 3200  ....hJMU..test2.
0000010: 0300 0000 0000 0000 0000                 ..........
```
So I think the best we can do is open (or gzip.open) the file, and check if calling `read` on it returns the empty string

**Comment by wasade** [comment]
Perhaps it is worth checking if its easy to just not produce the files?

**Comment by adamrp** [comment]
Thought that was coming from QIIME, but I might be mistaken. I'll have a look.

**Comment by wasade** [comment]
I thought we were generating from the demux files?

**Comment by adamrp** [comment]
Yep, just found that in qiita_ware/commands.py. Workig on it

**Comment by adamrp** [comment]
I think it will be easier just to change that code, so I'm going to close this PR.

**Pull request closed without merging**